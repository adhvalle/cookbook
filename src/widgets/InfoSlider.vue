<template>
  <div class="info-slider">
    <div class="info-slider__wrapper">
      <InfoPreview
        v-for="(item, index) in itemsList"
        :key="index"
        :info="item"
        :is-selected="index === selectedIndex"
        class="info-slider__item"
        data-test="info-slider__item"
        @click="$emit('on-click', index)"
      />
    </div>
  </div>
</template>

<script>
import InfoPreview from '@/components/InfoPreview.vue';

export default {
  name: 'InfoSlider',
  components: { InfoPreview },
  props: {
    itemsList: {
      type: Array,
      default: () => []
    },
    selectedIndex: {
      type: Number,
      default: 0
    }
  },
  emits: ['on-click']
}
</script>

<style>
.info-slider {
  margin-bottom: var(--size-base-x-3);
  overflow: scroll;
  background-color: var(--color-background-secondary);
  padding: var(--size-base-x-3);
}

@media (min-width: 1024px) {
  .info-slider {
    background-color: transparent;
    padding: 0;
  }
}

.info-slider__wrapper {
  display: flex;
  width: 100%;
}

@media (min-width: 1024px) {
  .info-slider__wrapper {
    display: block;
  }
}

.info-slider__item {
  min-width: 280px;
  margin-right: 16px;
  margin-left: 0;
}
</style>