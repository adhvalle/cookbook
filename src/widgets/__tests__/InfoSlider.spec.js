import { describe, it, expect } from 'vitest'

import { mount } from '@vue/test-utils'
import InfoSlider from '@/widgets/InfoSlider.vue'

describe('InfoSlider', () => {
  it('renders the component', () => {
    const wrapper = mountComponent()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('emits on-click event when one product is clicked', () => {
    const wrapper = mountComponent()
    wrapper.find('[data-test="info-slider__item"]').trigger('click')
    expect(wrapper.emitted('on-click')).toEqual([[0]])
  })

  const mountComponent = (props = {}) =>
    mount(InfoSlider, {
      props: { 
        itemsList: [
          {
            description: 'description',
            details: {
              area: 'area',
              category: 'category',
              video: 'H5SmjR-fxUs'
            },
            image: '1520081754.jpg',
            title: 'Title'
          },
          {
            description: 'description',
            details: {
              area: 'area',
              category: 'category',
              video: 'H5SmjR-fxUs'
            },
            image: '1520081754.jpg',
            title: 'Title'
          }
        ],
        ...props
      }
    })
})