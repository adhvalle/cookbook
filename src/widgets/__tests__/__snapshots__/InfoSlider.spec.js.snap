// Vitest Snapshot v1

exports[`InfoSlider > renders the component 1`] = `
"<div class=\\"info-slider\\">
  <div class=\\"info-slider__wrapper\\">
    <div class=\\"info-preview info-preview--selected info-slider__item\\" data-test=\\"info-slider__item\\">
      <div class=\\"info-preview__image\\"><img class=\\"info-preview__img-src\\" alt=\\"Title\\"></div>
      <div class=\\"info-preview__content\\">
        <h2 class=\\"info-preview__title\\">Title</h2>
        <p class=\\"info-preview__description\\">description</p>
      </div>
    </div>
    <div class=\\"info-preview info-slider__item\\" data-test=\\"info-slider__item\\">
      <div class=\\"info-preview__image\\"><img class=\\"info-preview__img-src\\" alt=\\"Title\\"></div>
      <div class=\\"info-preview__content\\">
        <h2 class=\\"info-preview__title\\">Title</h2>
        <p class=\\"info-preview__description\\">description</p>
      </div>
    </div>
  </div>
</div>"
`;
